drivers/spi/spi_mcux_lpspi.c

```diff
@@ -386,6 +438,40 @@ static int wait_dma_rx_tx_done(const struct device *dev)
 	}
 }
 
+static int spi_mcux_dma_rxtx_load(const struct device *dev,
+				size_t *dma_size)
+{
+	struct spi_mcux_data *lpspi_data = dev->data;
+	int result = 0;
+
+	/* Clear status flags */
+	lpspi_data->status_flags = 0U;
+	/* Load dma blocks of equal length */
+	*dma_size = MIN(lpspi_data->ctx.tx_len, lpspi_data->ctx.rx_len);
+	if (*dma_size == 0) {
+		*dma_size = MAX(lpspi_data->ctx.tx_len, lpspi_data->ctx.rx_len);
+	}
+
+	result = spi_mcux_dma_tx_load(dev, lpspi_data->ctx.tx_buf,
+					*dma_size);
+	if (!result) {
+
+		result = spi_mcux_dma_rx_load(dev, lpspi_data->ctx.rx_buf,
+						*dma_size);
+		if (!result) {
+			/* Start DMA */
+			result = dma_start(lpspi_data->dma_tx.dma_dev,
+						lpspi_data->dma_tx.channel);
+			if (!result) {
+				result = dma_start(lpspi_data->dma_rx.dma_dev,
+						lpspi_data->dma_rx.channel);
+			}
```

Comment:
please, do not nest so many if statements, just do `if (result) return result;` after each call, this way is confusing to read and change

Reply:
Changed, however this approach has only one return statement and exposes the dependency chain required for this function to return success.